File: node/publisher.js
const {PubSub} = require('@google-cloud/pubsub');

async function publish(projectId, topicName, data) {
    const pubsub = new PubSub({projectId});
    const topic = pubsub.topic(topicName);

    // https://stackoverflow.com/questions/59758539/what-is-the-correct-way-to-publish-to-gcp-pubsub-from-a-cloud-function
    topic.publish(Buffer.from(data));
}

var subject = 'Hello world!'
if (process.argv.length > 2 && process.argv[2] !== "") {
    subject = process.argv[2];
}

const data = JSON.stringify({
    'origin': 'node',
     subject
});

publish('gps-demo', 'demo-topic', data);

File: go/publisher/publisher.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"io"
	"os"

	"cloud.google.com/go/pubsub"
)

type Message struct {
	Origin  string `json:"origin"`
	Subject string `json:"subject"`
}

func publish(w io.Writer, projectID string, topicID string, msg Message) error {
	ctx := context.Background()
	client, err := pubsub.NewClient(ctx, projectID)
	if err != nil {
		return fmt.Errorf("pubsub: NewClient: %v", err)
	}
	defer client.Close()

	t := client.Topic(topicID)

	b, err := json.Marshal(msg)
	if err != nil {
		panic(err)
	}
	result := t.Publish(ctx, &pubsub.Message{
		Data: []byte(b),
	})

	// Block until the result is returned and a server-generated
	// ID is returned for the published message.
	id, err := result.Get(ctx)
	if err != nil {
		return fmt.Errorf("pubsub: result.Get: %v", err)
	}
	fmt.Fprintf(w, "Published a message; msg ID: %v\n", id)
	return nil
}

func main() {
	msg := Message{
		Origin:  "go",
		Subject: "Hello world",
	}
	if len(os.Args) > 1 && os.Args[1] != "" {
		msg.Subject = os.Args[1]
	}
	err := publish(os.Stdout, "gps-demo", "demo-topic", msg)
	if err != nil {
		fmt.Printf("ERROR: %s \n", err.Error())
	}
}
